src/spawn/arguments.ts

```diff
@@ -0,0 +1,64 @@
+const _ = require('underscore');
+const checkTypes = require('check-types');
+
+import { CanDeployOptions } from '../can-deploy';
+import { MessageOptions } from '../message';
+import { PublisherOptions } from '../publisher';
+import { ServiceOptions } from '../service';
+import { VerifierOptions } from '../verifier';
+
+export type SpawnArguments =
+  | CanDeployOptions[]
+  | MessageOptions
+  | PublisherOptions
+  | ServiceOptions
+  | VerifierOptions
+  | {};
+
+export const DEFAULT_ARG = 'DEFAULT';
+
+export class Arguments {
+  public fromSpawnArgs(
+    args: SpawnArguments,
+    mappings: { [id: string]: string },
+  ): string[] {
+    return _.chain(args instanceof Array ? args : [args])
+      .map((x: SpawnArguments) => this.createArgumentsFromObject(x, mappings))
+      .flatten()
+      .value();
+  }
+
+  public stringifySpawnArguments(obj: SpawnArguments): string {
```

Comment:
Should we remove `stringifySpawnArguments` altogether? It doesn't appear to be used anywhere anymore.